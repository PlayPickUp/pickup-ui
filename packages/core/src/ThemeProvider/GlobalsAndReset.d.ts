import { ReactNode } from "react";
import { DefaultTheme } from "../../../../types";
declare const GlobalsAndReset: import("react").ComponentType<
	Pick<
		{
			children: ReactNode | any;
		},
		"children"
	> &
		Partial<
			import("react-jss").WithStylesProps<
				(
					theme: DefaultTheme
				) => {
					"@global": {
						"html, body, div, span, applet, object, iframe, h1, h2, h3, h4, h5, h6, p, blockquote, pre, a, abbr, acronym, address, big, cite, code, del, dfn, em, img, ins, kbd, q, s, samp, small, strike, strong, sub, sup, tt, var, b, u, i, center, dl, dt, dd, ol, ul, li, fieldset, form, label, legend, table, caption, tbody, tfoot, thead, tr, th, td, article, aside, canvas, details, embed, figure, figcaption, footer, header, hgroup, menu, nav, output, ruby, section, summary, time, mark, audio, video": {
							margin: string;
							padding: string;
							border: string;
							fontSize: string;
							font: string;
							verticalAlign: string;
						};
						"article, aside, details, figcaption, figure, footer, header, hgroup, menu, nav, section": {
							display: string;
						};
						body: {
							lineHeight: string;
						};
						"ol, ul": {
							listStyle: string;
						};
						"blockquote, q": {
							quotes: string;
						};
						"blockquote:before, blockquote:after, q:before, q:after": {
							content: string;
						};
						table: {
							borderCollapse: string;
							borderSpacing: string;
						};
						"html, body": {
							width: string;
							minHeight: string;
							fontFamily: string;
							fontSize: import("react").ReactText;
							fontWeight: number;
							color: string;
							backgroundColor: string;
							"-webkit-font-smoothing": string;
							"-moz-osx-font-smoothing": string;
							margin: string;
							padding: string;
						};
						html: {
							boxSizing: string;
							background: string;
						};
						"*, *:before, *:after": {
							boxSizing: string;
						};
						button: {
							display: string;
							"-webkit-appearance": string;
							"-moz-appearance": string;
							userSelect: string;
							textDecoration: string;
							fontFamily: string;
							fontSize: string;
							color: string;
							cursor: string;
							border: string;
							backgroundColor: string;
							borderColor: string;
							padding: string;
							margin: string;
							"&:disabled": {
								cursor: string;
							};
						};
					};
				}
			>
		> & {
			innerRef?:
				| import("react").RefObject<any>
				| ((instance: any) => void)
				| undefined;
		}
>;
export default GlobalsAndReset;
//# sourceMappingURL=GlobalsAndReset.d.ts.map
