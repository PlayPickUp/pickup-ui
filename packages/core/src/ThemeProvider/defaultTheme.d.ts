import { DefaultTheme } from "../../../../types";
export declare const defaultTheme: DefaultTheme;
//# sourceMappingURL=defaultTheme.d.ts.map
