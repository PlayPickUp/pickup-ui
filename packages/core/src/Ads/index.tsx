import React from "react";
import { createUseStyles } from "react-jss";
import { AdZoneProps, DefaultTheme } from "../types";

const useStyles = createUseStyles((theme: DefaultTheme) => ({
  foregroundImage: {
    maxWidth: "100%",
    maxHeight: "100%",
    display: "block",
    objectFit: "contain",
    margin: "auto",
  },
  foregroundImageContainer: {
    width: "33%",
    height: "90%",
    zIndex: "1",
  },
  copy: {
    display: "flex",
    maxWidth: "60%",
    flexDirection: "row",
    "& > :first-child": {
      marginRight: 5,
    },
  },
  container: {
    display: "flex",
    flexDirection: "row",
    justifyContent: "space-around",
    alignItems: "center",
    textAlign: "center",
    borderRadius: "4px",
    backgroundSize: `contain`,
    backgroundRepeat: `no-repeat`,
    backgroundColor: theme.colors.grey.light,
  },
}));

const Ad: React.FC<AdZoneProps> = ({ width, height, adProps }) => {
  const classes = useStyles();
  return (
    <div data-testid="ad">
      {adProps.url ? (
        <a href={adProps.url}>
          <div
            style={{
              width: `${width}px`,
              height: `${height}px`,
              backgroundImage: `url(${adProps.backgroundImage}`,
            }}
            className={classes.container}
          >
            {adProps.foregroundImage ? (
              <div className={classes.foregroundImageContainer}>
                <img
                  className={classes.foregroundImage}
                  src={adProps.foregroundImage}
                  alt={adProps.name}
                />
              </div>
            ) : null}
            {adProps.copy ? (
              <div
                className={classes.copy}
                dangerouslySetInnerHTML={{
                  __html: adProps.copy,
                }}
              ></div>
            ) : null}
          </div>
        </a>
      ) : (
        <div
          style={{
            width: `${width}px`,
            height: `${height}px`,
            backgroundImage: `url(${adProps.backgroundImage}`,
            backgroundSize: `contain`,
            backgroundRepeat: `no-repeat`,
            backgroundColor: `${defaultTheme.colors.grey.light}`,
          }}
          className={classes.container}
        >
          {adProps.foregroundImage ? (
            <div className={classes.foregroundImageContainer}>
              <img
                className={classes.foregroundImage}
                src={adProps.foregroundImage}
                alt={adProps.name}
              />
            </div>
          ) : null}
          {adProps.copy ? (
            <div
              className={classes.copy}
              dangerouslySetInnerHTML={{
                __html: adProps.copy,
              }}
            ></div>
          ) : null}
        </div>
      )}
    </div>
  );
};

export default Ad;
