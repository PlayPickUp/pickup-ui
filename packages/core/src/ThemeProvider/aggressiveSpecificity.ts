/* This is based on the unsupported jss-increase-specificity plugin at https://github.com/iamstarkov/jss-increase-specificity
JSS's original createGenerateId has been adapted here to use logic from that plugin. /*

// From the original plugin:
/**
 * use :not(#\20), :not(.\20) and :not(\20) instead of generating unlikely
 * appearing ids…
 * — twitter.com/subzey/status/829050478721896448
 * Rationale: \20 is a css escape for U+0020 Space, and neither classname,
 * nor id, nor tagname can contain a space
 * — twitter.com/subzey/status/829051085885153280
 */

import { Rule, CreateGenerateId } from "jss";
import { JssStyleSheet } from "../types";

var selector = ":not(#\\20)";
const repeat = 3;
const maxRules = 1e10;

export const increaseSpecificity: CreateGenerateId = (options = {}) => {
  let ruleCounter = 0;
  console.log({ options });
  return (rule: Rule, sheet?: JssStyleSheet): string => {
    ruleCounter += 1;
    if (ruleCounter > maxRules) {
      console.warn(
        false,
        `[JSS] You might have a memory leak. Rule counter is at ${ruleCounter}.`
      );
    }
    const jssId = String(sheet.options.jss.id) ?? "";
    const prefix = sheet.options.classNamePrefix ?? "";

    let suffix = "";
    for (let i = 0; i < repeat; i++) {
      suffix += selector;
    }

    return `${prefix + rule.key}${
      jssId ? `-${jssId}` : ""
    }-${ruleCounter}${suffix}`;
  };
};
