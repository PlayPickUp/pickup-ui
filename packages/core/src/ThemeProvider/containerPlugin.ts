/*
 *  Adds container params to all selectors
 *  Example: 'h1' becomes 'body #PickUpUI h1'
 */

import { Plugin } from "jss";

export function containUI(withReset: boolean, container?: string): Plugin {
  const prefix = container ? container : "body #PickUpUI ";

  // 'rule' is left untyped because the Rule import from jss is not recognizing all child properties
  function onProcessRule(rule): Plugin["onProcessRule"] {
    const parent = rule.options.parent;
    if (
      rule.type !== "style" ||
      rule.selectorText.includes(prefix) ||
      (parent && parent.type === "keyframes") ||
      (withReset && parent && parent.key === "@global")
    )
      return;

    if (rule.selectorText.includes(",")) {
      const allSelectors = rule.selectorText.split(",");
      const transformedSelectors = allSelectors.map((selector: string) =>
        selector.includes(prefix) ? selector : prefix + selector
      );
      rule.selectorText = transformedSelectors.join(",");
    } else {
      rule.selectorText = prefix + rule.selectorText;
    }
  }
  return { onProcessRule };
}
