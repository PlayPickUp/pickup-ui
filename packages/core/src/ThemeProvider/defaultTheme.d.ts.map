{"version":3,"file":"defaultTheme.d.ts","sourceRoot":"","sources":["defaultTheme.ts"],"names":[],"mappings":"AAAA,OAAO,EAMN,YAAY,EAWZ,MAAM,mBAAmB,CAAC;AAqJ3B,eAAO,MAAM,YAAY,EAAE,YAU1B,CAAC"}