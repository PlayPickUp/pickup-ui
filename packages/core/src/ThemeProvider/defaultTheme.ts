import {
  Breakpoints,
  CreateShadow,
  DefaultGradients,
  DefaultShadows,
  DefaultSpacing,
  DefaultTheme,
  DefaultThemeColors,
  DefaultTypography,
  DefaultZIndex,
  FontFamilies,
  GreenMap,
  GreyMap,
  MediaQuery,
  PurpleMap,
  RedMap,
  ResponsiveFonts,
} from "../types";

// Color me PickUp âš¡
const purple: PurpleMap = {
  light: "#F5F0FF",
  lightBase: "#AFA0FF",
  base: "#7A60FF",
  dark: "#5E24FF",
  transparent: "#EAE5FF",
};

const green: GreenMap = {
  light: "#EAFDF7",
  base: "#25DA95",
};

const grey: GreyMap = {
  light: "#F3F2F5",
  lightBase: "#DDDDDD",
  base: "#9D9D9E",
  dark: "#615E66",
  medium: "#DFDAE6",
};

const red: RedMap = {
  light: "#FFF1F4",
  base: "#FF1244",
};

const colors: DefaultThemeColors = {
  primary: { ...purple },
  secondary: { ...grey },
  green,
  purple,
  grey,
  red,
  black: "#2C2933",
  white: "#FFF",
  success: green.base,
  error: red.base,
};

// Font and Typography
const fontFamilies: FontFamilies = {
  body: "'Inter', Helvetica, sans-serif",
  headline: "'Ringside-Condensed', Helvetica, sans-serif",
};

const fontStyles: ResponsiveFonts = {
  desktop: {
    title: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 48,
      lineHeight: "54px",
      letterSpacing: "0.1px",
      textTransform: "uppercase",
      color: "inherit",
    },
    heading2: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 36,
      lineHeight: "40px",
      letterSpacing: "0.1px",
      textTransform: "uppercase",
      color: "inherit",
    },
    heading3: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 28,
      lineHeight: "34px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
    heading4: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 22,
      lineHeight: "28px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
    heading5: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 16,
      lineHeight: "22px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
    heading6: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 13,
      lineHeight: "16px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
    body: {
      fontFamily: fontFamilies.body,
      fontStyle: "normal",
      fontWeight: "normal",
      fontSize: 18,
      lineHeight: "30px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
    body2: {
      fontFamily: fontFamilies.body,
      fontStyle: "normal",
      fontWeight: "normal",
      fontSize: 13,
      lineHeight: "20px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
  },
  mobile: {
    title: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 34,
      lineHeight: "38px",
      letterSpacing: "0.1px",
      textTransform: "uppercase",
      color: "inherit",
    },
    heading2: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 28,
      lineHeight: "34px",
      letterSpacing: "0.1px",
      textTransform: "uppercase",
      color: "inherit",
    },
    heading3: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 22,
      lineHeight: "26px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
    heading4: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 16,
      lineHeight: "18px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
    heading5: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 13,
      lineHeight: "15px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
    heading6: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 11,
      lineHeight: "13px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
    body: {
      fontFamily: fontFamilies.body,
      fontStyle: "normal",
      fontWeight: "normal",
      fontSize: 15,
      lineHeight: "24px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
    body2: {
      fontFamily: fontFamilies.body,
      fontStyle: "normal",
      fontWeight: "normal",
      fontSize: 12,
      lineHeight: "18px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
  },
};

const typography: DefaultTypography = {
  fontFamilies,
  fontStyles,
};

// Fancy ass Material UI inspired shadows
const shadowKeyUmbraOpacity = 0.2;
const shadowKeyPenumbraOpacity = 0.14;
const shadowAmbientShadowOpacity = 0.12;

const createShadow: CreateShadow = (...px) => {
  return [
    `${px[0]}px ${px[1]}px ${px[2]}px ${px[3]}px rgba(0, 0, 0, ${shadowKeyUmbraOpacity})`,
    `${px[4]}px ${px[5]}px ${px[6]}px ${px[7]}px rgba(0, 0, 0, ${shadowKeyPenumbraOpacity})`,
    `${px[8]}px ${px[9]}px ${px[10]}px ${px[11]}px rgba(0, 0, 0, ${shadowAmbientShadowOpacity})`,
  ].join(",");
};

const shadows: DefaultShadows = [
  createShadow(0, 1, 3, 0, 0, 1, 1, 0, 0, 2, 1, -1),
  createShadow(0, 1, 5, 0, 0, 2, 2, 0, 0, 3, 1, -2),
  createShadow(0, 1, 8, 0, 0, 3, 4, 0, 0, 3, 3, -2),
  createShadow(0, 2, 4, -1, 0, 4, 5, 0, 0, 1, 10, 0),
  createShadow(0, 3, 5, -1, 0, 5, 8, 0, 0, 1, 14, 0),
  createShadow(0, 3, 5, -1, 0, 6, 10, 0, 0, 1, 18, 0),
  createShadow(0, 4, 5, -2, 0, 7, 10, 1, 0, 2, 16, 1),
  createShadow(0, 5, 5, -3, 0, 8, 10, 1, 0, 3, 14, 2),
  createShadow(0, 5, 6, -3, 0, 9, 12, 1, 0, 3, 16, 2),
  createShadow(0, 6, 6, -3, 0, 10, 14, 1, 0, 4, 18, 3),
  createShadow(0, 6, 7, -4, 0, 11, 15, 1, 0, 4, 20, 3),
  createShadow(0, 7, 8, -4, 0, 12, 17, 2, 0, 5, 22, 4),
  createShadow(0, 7, 8, -4, 0, 13, 19, 2, 0, 5, 24, 4),
  createShadow(0, 7, 9, -4, 0, 14, 21, 2, 0, 5, 26, 4),
  createShadow(0, 8, 9, -5, 0, 15, 22, 2, 0, 6, 28, 5),
  createShadow(0, 8, 10, -5, 0, 16, 24, 2, 0, 6, 30, 5),
  createShadow(0, 8, 11, -5, 0, 17, 26, 2, 0, 6, 32, 5),
  createShadow(0, 9, 11, -5, 0, 18, 28, 2, 0, 7, 34, 6),
  createShadow(0, 9, 12, -6, 0, 19, 29, 2, 0, 7, 36, 6),
  createShadow(0, 10, 13, -6, 0, 20, 31, 3, 0, 8, 38, 7),
  createShadow(0, 10, 13, -6, 0, 21, 33, 3, 0, 8, 40, 7),
  createShadow(0, 10, 14, -6, 0, 22, 35, 3, 0, 8, 42, 7),
  createShadow(0, 11, 14, -7, 0, 23, 36, 3, 0, 9, 44, 8),
  createShadow(0, 11, 15, -7, 0, 24, 38, 3, 0, 9, 46, 8),
];

// Base Unit Spacings
const spacing: DefaultSpacing = {
  base: 4, // ðŸš¨ changing this value (4) could have VERY BAD CONCEQUENCES! BE VERY SURE, MAYBE YOU JUST NEED TO ADD A VALUE? ðŸ¤”
};

// Default zIndex values
const zIndex: DefaultZIndex = {
  overlay: 999,
};

// Default Border Radius
const borderRadius = 4;

// Media Queries
const mediaQuery: MediaQuery = (minWidth: number) => {
  return `@media (min-width: ${minWidth}px)`;
};

// Breakpoints
const breakpoints: Breakpoints = {
  small: 787,
  medium: 1029,
  large: 1280,
};

// Default Gradient Values
const gradients: DefaultGradients = {
  main: `linear-gradient(112.25deg, #914EFF -20.38%, #5D1BFF 122.56%)`,
  hover: `linear-gradient(112.25deg, #914EFF 35.38%, #5D1BFF 122.56%)`,
  disabled:
    "linear-gradient(112.25deg, rgba(145,78,255,0.2) -20.38%, rgba(93,27,255,0.2) 122.56%)",
};

export const defaultTheme: DefaultTheme = {
  borderRadius,
  breakpoints,
  colors,
  gradients,
  mediaQuery,
  shadows,
  spacing,
  typography,
  zIndex,
};
