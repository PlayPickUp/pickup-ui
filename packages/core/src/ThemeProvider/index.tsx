import React from "react";
import { jss, JssProvider, ThemeProvider as JSSThemeProvider } from "react-jss";
import { containUI } from "./containerPlugin";
import merge from "lodash/merge";

import { defaultTheme } from "./defaultTheme";

import { ThemeProviderProps } from "../types";
import GlobalsAndReset from "./GlobalsAndReset";

function generateRandom(digitCount) {
  return Math.floor(
      Math.pow(10, digitCount - 1) +
      Math.random() * 9 * Math.pow(10, digitCount - 1)
  );
}

const ThemeProvider: React.FC<ThemeProviderProps> = ({
  children,
  withReset = true,
  theme,
  ...rest
}) => {
  jss.use(containUI(withReset));
  const mergedTheme = theme ? merge(defaultTheme, theme) : defaultTheme;

  const random = generateRandom(5)

  return (
    <div id="PickUpUI">
      <JssProvider jss={jss} classNamePrefix={`PU${random}--`}>
        <JSSThemeProvider theme={mergedTheme} {...rest}>
          <GlobalsAndReset>{children}</GlobalsAndReset>
        </JSSThemeProvider>
      </JssProvider>
    </div>
  );
};

export default ThemeProvider;
