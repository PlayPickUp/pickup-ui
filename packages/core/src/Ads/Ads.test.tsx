import React from "react";
import DOMPurify from "dompurify";
import { render } from "@testing-library/react";

import ThemeProvider from "../ThemeProvider";
import Ads from "./index";

const AdElement = (
  <ThemeProvider>
    <Ads
      height={100}
      width={300}
      name={"HelloWorld Ad"}
      url={"https://www.playpickup.com"}
      backgroundImage={
        "https://images.ctfassets.net/vr34jcb0tstv/42KLMVnUxtYHYkBNfPh9CH/22f587c10adcd33faa6b3325f1096de0/BetMGM_200_Suns.png"
      }
      foregroundImage={
        "https://images.ctfassets.net/vr34jcb0tstv/2XgjxzpUehZpgDwyeBTyVw/6f4f124a291d84da963b5491e4dc20b3/hero.svg"
      }
      copy={
        '<script id="data" type="application/json">{"userId":1234,"userName":"John Doe","memberSince":"2000-01-01T00:00:00.000Z"}</script><h1>Hello World!</h1>'
      }
    />
  </ThemeProvider>
);

test("Renders and matches snapshot", () => {
  Math.random = jest.fn(() => 1); // <--- This is the key, overriding the system's Math.random function
  const { container } = render(AdElement);
  expect(container).toMatchSnapshot();
});

test("Renders image for foreground", () => {
  const { container } = render(AdElement);
  expect(container.getElementsByTagName("img")[0].getAttribute("src")).toEqual(
    "https://images.ctfassets.net/vr34jcb0tstv/2XgjxzpUehZpgDwyeBTyVw/6f4f124a291d84da963b5491e4dc20b3/hero.svg"
  );
});

test("Renders image for background", () => {
  const container = render(AdElement);
  const background = container.queryByTestId("background"); // get the background div
  expect(background).toHaveProperty("style.backgroundImage");
});

test("Renders Copy", () => {
  const { getByText } = render(AdElement);
  expect(getByText("Hello World!")).toBeTruthy();
});

test("Copy is purified", () => {
  const { container } = DOMPurify(render(AdElement));
  expect(container).toMatchSnapshot();
});
