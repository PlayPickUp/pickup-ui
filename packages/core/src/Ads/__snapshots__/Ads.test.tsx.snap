// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders and matches snapshot 1`] = `
<div>
  <div
    id="PickUpUI"
  >
    <div
      data-testid="ad"
    >
      <div
        class="PU100000--container-0-2-5"
        style="width: 300px; height: 100px;"
      >
        <div
          class="PU100000--foregroundImageContainer-0-2-3"
        >
          <img
            alt="HelloWorld Ad"
            class="PU100000--foregroundImage-0-2-2"
            src="https://images.ctfassets.net/vr34jcb0tstv/2XgjxzpUehZpgDwyeBTyVw/6f4f124a291d84da963b5491e4dc20b3/hero.svg"
          />
        </div>
        <div
          class="PU100000--copy-0-2-4"
        >
          <h1>
            Hello World!
          </h1>
        </div>
      </div>
    </div>
  </div>
</div>
`;
